lib/instance-detail-helper.js

```diff
@@ -0,0 +1,527 @@
+        name: resource.collection
+      };
+    });
+
+  collections = _.map(collections, parseCollection);
+  debug('allowed collections', collections);
+  done(null, collections);
+}
+
+function getDatabaseCollections(db, done) {
+  debug('getDatabaseCollections...');
+
+  const spec = {};
+
+  /**
+   * @note: Durran: For some reason the listCollections call does not take into
+   *  account the read preference that was set on the db instance - it only looks
+   *  in the passed options: https://github.com/mongodb/node-mongodb-native/blob/2.2/lib/db.js#L671
+   */
+  const rp = db.s ? db.s.readPreference : ReadPreference.PRIMARY;
+  const options = { readPreference: rp };
+
+  db.listCollections(spec, options).toArray(function(err, res) {
+    if (err) {
+      if (isNotAuthorized(err)) {
+        // if the error is that the user is not authorized, silently ignore it
+        // and return an empty list
+        debug('not allowed to run `listCollections` command on %s, returning'
+          + ' empty result [].', db.databaseName);
+        return done(null, []);
+      }
+      // the command failed for another reason, report the error
+      debug('listCollections failed', err);
+      err.command = 'listCollections';
+      return done(err);
+    }
+    done(null, _.map(res, function(d) {
+      d.db = db.databaseName;
+      return parseCollection(d);
+    }));
+  });
+}
+
+function getCollections(results, done) {
+  // var db = results.db;
+
+  // concat
+  let collections = [].concat.apply(
+    results.listCollections, results.allowedCollections);
+  // de-dupe based on _id
+  collections = _.uniq(collections, '_id');
+
+  // @todo filter the ones that we can "count on"
+  // async.filter(collections, function(collection, callback) {
+  //   db.db(collection.database).collection(collection.name).count(function(err) {
+  //     if (err) {
+  //       return callback(false);
+  //     }
+  //     callback(true);
+  //   });
+  // }, function(countableCollections) {
+  //   done(null, countableCollections);
+  // });
+  done(null, collections);
+}
+
+function listCollections(results, done) {
+  const db = results.db;
+  const databases = results.databases;
+
+  // merge and de-dupe databases
+  const tasks = _.map(databases, function(_db) {
+    return getDatabaseCollections.bind(null, db.db(_db.name));
+  });
+
+  async.parallel(tasks, function(err, res) {
+    if (err) {
+      debug('listCollections failed', err);
+      return done(err);
+    }
+    done(null, _.flatten(res));
+  });
+}
+
+function getHierarchy(results, done) {
+  const databases = results.databases;
+  const collections = _.groupBy(results.collections, function(collection) {
+    return collection.database;
+  });
+  _.each(databases, function(db) {
+    db.collections = collections[db.name] || [];
+  });
+  done();
+}
+
+
+function attach(anything, done) {
+  done(null, anything);
+}
+
+/**
+ * pass in a connection model and get instance details back.
```

Comment:
Not sure if the comment was like that in #fetch already, but I think we are not passing in a connection model anymore, correct? Can you update it?